client/lib/wp/sync-handler/index.js

```diff
@@ -121,26 +163,31 @@ export class SyncHandler {
 			 * WP.com server-side response.
 			 *
 			 * @param {Object} serverResponse - server response object
+			 * @returns {Promise} - promise that resolves with server response object
 			 */
 			const cacheResponse = serverResponse => {
-				// get response object without _headers property
-				let responseWithoutHeaders = Object.assign( {}, serverResponse );
-				delete responseWithoutHeaders._headers;
-
-				let storingData = {
-					__sync: {
-						key,
-						synced: Date.now(),
-						syncing: false
-					},
-					body: responseWithoutHeaders,
-					params: reqParams
-				};
-
-				// add/override gotten data from server-side
-				this.storeRecord( key, storingData ).catch( err => {
-					// @TODO error handling
-					warn( err );
+				return new Promise( ( resolve ) => {
+					// get response object without _headers property
+					let responseWithoutHeaders = Object.assign( {}, serverResponse );
+					delete responseWithoutHeaders._headers;
+
+					let storingData = {
+						__sync: {
+							key,
+							synced: Date.now(),
+							syncing: false
+						},
+						body: responseWithoutHeaders,
+						params: reqParams
+					};
+
+					// add/override gotten data from server-side
+					this.storeRecord( key, storingData ).catch( err => {
+						// @TODO error handling
+						warn( err );
+					} );
+
+					resolve( serverResponse );
```

Comment:
> IMO the error handling when the server request fails should be done before to try to store the local data. Thus combining both ideas I thought in something like the following ....

@retrofox As I understand that is already the case, because `networkFetch` returns a promise, and upon getting a response/failure it either does `resolve` (and therefore runs `cacheResponse`) or it runs `reject` and therefore runs `networkErrorHandler` (instead of `cacheResponse`).